hana/monitoring.sls

```diff
@@ -0,0 +1,104 @@
+{%- from "hana/map.jinja" import hana with context -%}
+
+{% set pydbapi_output_dir = '/tmp/pydbapi' %}
+
+{% for node in hana.nodes if node.host == grains['host'] %}
+
+{% if node.secondary is not defined %}
+{% set exporter = node.exporter %}
```

Comment:
Hmm, this is a problem that is present everywhere in these salt formulas: we cannot validate group of fields so that they make logical sense together.

In the other formulas we don't have all this granularity: if we enable monitoring it just works, we don't need to specify anything else to configure the exporters correctly, it's done by the state. That said, the HANA exporter is a bit different and more complicated, as we discussed already, so I guess that approach doesn't apply here.

That said, we could mitigate this by putting all of the rest of the salt state under an `if` condition on the `exporter` variable not being `null`, similarly to how it was being done before.